src/Rox/Core/XPack/Network/StateSender.php

```diff
@@ -150,10 +159,29 @@ private function _serializeCustomProperties()
         $allCustomProperties = $this->_customPropertyRepository->getAllCustomProperties();
         $keys = array_keys($allCustomProperties);
         sort($keys);
-        return array_map(function ($key) use ($allCustomProperties) {
-            $value = (string)$allCustomProperties[$key];
+        $mapped = array_map(function ($key) use ($allCustomProperties) {
+            $value = (string) $allCustomProperties[$key];
+
+            if (!$this->_isCBP() && $allCustomProperties[$key]->getType()->getExternalType() === "DateTime") {
+                return null;
+            }
+
             return json_decode($value, true);
         }, $keys);
+
+        return array_filter($mapped, function ($v) {
+            return $v !== null;
+        });
+
+    }
+
+    /**
+     * @return bool
+     */
+    private function _isCBP()
+    {
+        $uuidApiKeyPattern = "/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i";
+        return preg_match($uuidApiKeyPattern, $this->_deviceProperties->getRolloutKey());
```

Comment:
this should be the same test as in rox.php
if needed, it should be moved to core